### PR TITLE
[reigns] Fix/fres 1458 fix demo no cooldown

### PR DESCRIPTION
**Pull request by maxired**
this PR fixes the fact that currently when using the public demo.json files, some cards can be reshown to the user.
the original can definition had not value for cooldown, which was wrongly converted to an empty string.

this PR fixes it, and adds some validations.

### CODE REVIEW

File: elements/reigns/src/features/game/validateGameDefinition.test.ts [hunk 1]
```diff
@@ -114,6 +114,81 @@ describe("validateGameDefinition", () => {
         ])
       ).not.toThrow();
     });
+
+    it('should not thrown if cooldown is a positive number', () => {
```
Comment (staff0rd) [inline elements/reigns/src/features/game/validateGameDefinition.test.ts:118]:
```suggestion
    it('should not throw if cooldown is a positive number', () => {
```

File: elements/reigns/src/features/game/validateGameDefinition.test.ts [hunk 2]
```diff
@@ -114,6 +114,81 @@ describe("validateGameDefinition", () => {
...
+          card: "some card",
+          weight: 1,
+          cooldown: 10,
+        } as Card,
+      ])
+    ).not.toThrow()
+
+    })
+
+    it('should not thrown if cooldown is a 0', () => {
```
Comment (staff0rd) [inline elements/reigns/src/features/game/validateGameDefinition.test.ts:131]:
```suggestion
    it('should not throw if cooldown is a 0', () => {
```

Comment (staff0rd) [reply elements/reigns/src/features/game/validateGameDefinition.test.ts:131]: and the rest :)

File: elements/reigns/src/features/game/validateGameDefinition.ts [hunk 3]
```diff
@@ -48,6 +48,20 @@ export const validateCards = (cards: Card[] | undefined): Card[] => {
...
+    if (card.cooldown !== null && card.cooldown !== undefined) {
+      if (!Number.isInteger(card.cooldown)) {
+        throw new Error(
+          `Card ${i + 1} is invalid, cooldown must be an interger`
+        );
+      }
+
+      if (card.cooldown < 0) {
+        throw new Error(
+          `Card ${i + 1} is invalid, cooldown must be positve of equal to 0`
```
Comment (staff0rd) [inline elements/reigns/src/features/game/validateGameDefinition.ts:60]:
```suggestion
          `Card ${i + 1} is invalid, cooldown must be positve or equal to 0`
```

File: elements/reigns/src/public.spec.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            expect(()=> validateCards(demo.cards as any as Card[])).not.toThrow()
+        })
+    })
+
+    describe('gdpr.json validation',() => {
+
+        it('should not throw', () => {
+            expect(()=> validateCards(gdpr.cards)).not.toThrow()
+        })
+    })
```
Comment (staff0rd) [inline elements/reigns/src/public.spec.ts:21]: dupe: https://github.com/fres-co/fresco-community/blob/3f4c5133c8fed52f816b665b20a8ffcc03ac7317/elements/reigns/src/features/game/validateGameDefinition.test.ts#L188-L192

Comment (maxired) [reply elements/reigns/src/public.spec.ts:21]: thanks. then I guess I' will add the demo here ;-)

**Pull request merged**